pkg/cmd/experimental/syncgroups/grouplister.go

```diff
@@ -0,0 +1,109 @@
+package syncgroups
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/util"
+
+	ldaputil "github.com/openshift/origin/pkg/auth/authenticator/password/ldappassword/util"
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/config"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/interfaces"
+	userapi "github.com/openshift/origin/pkg/user/api"
+)
+
+// NewAllOpenShiftGroupLister returns a new AllLocalGroupLister
+func NewAllOpenShiftGroupLister(ldapSyncConfig config.LDAPSyncCommonConfig) interfaces.LDAPGroupLister {
```

Comment:
Accept the arguments you need.  In this case, the `dapSyncConfig.CLIOptions.GroupClient` and the `ldapSyncConfig.ServerConfig`